### DISCUSSION TITLE
Ports stuck in polling mode

### DISCUSSION THREAD

**Discussion by blackout565** [mvp/uhubctl#412] [Q&A]
Hello,

First of all thank you for providing this tool. 

I am trying to use it to make a raspberry pi device that controls a mega4 board to test the reliability of USB sticks.

I am using a dd command that targets the stick, and while dd is running I power off the ports from the hub.

The script I am using looks like this:

    ```python
    cmd_list = [[f"dd if=/dev/urandom of=/dev/{str(child_name)} bs=1M count=950"] for child_name in child_names]
    procs_list = [subprocess.Popen(cmd,shell=True, stdout=subprocess.PIPE).pid for cmd in cmd_list]
    for proc in procs_list:
        proc
    time.sleep(65)
    os.system("uhubctl -l 2-2 -a off")
    time.sleep(1)
    os.system("uhubctl -l 2-2 -a on")
    ```

After I go trough a few of these cycles when I do `uhubctl` I get:

Current status for hub 2-2 [2109:0817 VIA Labs, Inc. USB3.0 Hub, USB 3.10, 4 ports, ppps]
  Port 1: 02e0 power 5gbps Polling <---
  Port 2: 0203 power 5gbps U0 enable connect [1370:0502 Swissbit USB Flash Drive 6016852492000050]
  Port 3: 02e0 power 5gbps Polling <---
  Port 4: 02a0 power 5gbps Rx.Detect

I cannot access the USB sticks that are getting into the `polling` status until I reboot the raspberry pi or reconnect the hub.

The interesting thing is that I do not get any such issues if I am using USB 2.0 sticks, this only happens with USB 3.0 ones.
If I manually replug a USB 3.0 stick into a port with the "polling" state, it does not help. If I put a USB2.0 in such a port, it will get recognized.

If I simply power cycle the ports without any transfers in the background, everything works fine.

Has anyone got into this? Do you have any idea how I could get the ports unstuck from this `polling state` without having to reboot the pi or unplug the hub?

Many thanks in advance.

**Comment by mvp** [comment]
Since manual replugging does not help to fix this problem, I believe it means Linux kernel USB driver bug. Perhaps it's worth reporting this to Linux USB mailing list.
Can you try running your script on other OS, e.g. MacOS or FreeBSD?
